metrotranscom/doorway
# feat: add readable demographics data

# Pull Request Template

## Issue Overview

This PR addresses #569.


## Description

This PR adds functions for gender, sexualOrientation, spokenLanguage and howDidYouHear that convert the [form keys](https://github.com/metrotranscom/doorway/blob/6436c94d659333d5d6bacc10843c0fa213681b45/shared-helpers/src/utilities/formKeys.ts#L174) into a human readable format when exported. It also updates the race function keys. 

## How Can This Be Tested/Reviewed?

To review, first add or edit an application that answers all of the demographic questions. Particular details to note:
- Spoken language has a "Not Listed" option where the user can specify another language.
- Users can select multiple responses for the howDidYouHear question.

Then, export the data to see that the demographic data is human readable. 

## Checklist:


## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue, and ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.
